File: BlockPartyWindowsStore/Gameplay/BlockParticle.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlockPartyWindowsStore.Gameplay
{
    class BlockParticle
    {
        BlockRain blockRain;
        Rectangle rectangle;
        Vector2 velocity;
        Vector2 acceleration;
        Texture2D texture;
        public bool Active = true;
        static Random random = new Random();

        public BlockParticle(BlockRain blockRain)
        {
            this.blockRain = blockRain;

            int width = random.Next(blockRain.Screen.ScreenManager.Game.WorldViewport.Width > blockRain.Screen.ScreenManager.Game.WorldViewport.Height ?
                blockRain.Screen.ScreenManager.Game.WorldViewport.Height / 5 :
                blockRain.Screen.ScreenManager.Game.WorldViewport.Width / 5);
            int height = width;
            int x = random.Next(blockRain.Screen.ScreenManager.Game.WorldViewport.Width);
            int y = -1 * height;

            rectangle = new Rectangle(x, y, width, height);
            velocity = Vector2.Zero;
            acceleration = new Vector2(0, 0.25f);
            texture = blockRain.Textures[random.Next(blockRain.Textures.Count)];
        }

        public void Update(GameTime gameTime)
        {
            if (Active)
            {
                velocity += acceleration;
                rectangle.Y += (int)velocity.Y;

                if (rectangle.Y >= blockRain.Screen.ScreenManager.Game.WorldViewport.Height)
                {
                    Active = false;
                }
            }
        }

        public void Draw(GameTime gameTime)
        {
            if (Active)
            {
                blockRain.Screen.ScreenManager.Game.GraphicsManager.SpriteBatch.Draw(texture, rectangle, Color.White);
            }
        }
    }
}

File: BlockPartyWebRole/Client/BlockParticle.js
/// <reference path="Vector2.ts" />
/// <reference path="Rectangle.ts" />
/// <reference path="BlockRain.ts" />
var BlockParticle = (function () {
    function BlockParticle(blockRain) {
        this.Active = true;
        this.blockRain = blockRain;

        var width = Math.random() * (this.blockRain.Screen.ScreenManager.Game.WorldViewport.Width > this.blockRain.Screen.ScreenManager.Game.WorldViewport.Height ? this.blockRain.Screen.ScreenManager.Game.WorldViewport.Height / 5 : this.blockRain.Screen.ScreenManager.Game.WorldViewport.Width / 5);
        var height = width;
        var x = Math.random() * this.blockRain.Screen.ScreenManager.Game.WorldViewport.Width;
        var y = -1 * height;

        this.rectangle = new Rectangle(x, y, width, height);
        this.velocity = Vector2.Zero;
        this.acceleration = new Vector2(0, 0.25);
        this.image = this.blockRain.Images[Math.floor(Math.random() * this.blockRain.Images.length)];
    }
    BlockParticle.prototype.Update = function (gameTime) {
        if (this.Active) {
            this.velocity = this.velocity.AddVector(this.acceleration);
            this.rectangle.Y += this.velocity.Y;

            if (this.rectangle.Y >= this.blockRain.Screen.ScreenManager.Game.WorldViewport.Height) {
                this.Active = false;
            }
        }
    };

    BlockParticle.prototype.Draw = function (gameTime) {
        if (this.Active) {
            this.blockRain.Screen.ScreenManager.Game.GraphicsManager.Draw(this.image, this.rectangle, "white");
        }
    };
    return BlockParticle;
})();
